docs: update CONTRIBUTING.md

joshuajames-smith opened the pull request:
## Description

Large documentation update to CONTRIBUTING.md which has absorbed and expanded upon the previous version to help improve the gliff.ai GitHub Community Health Files.

joshuajames-smith committed:
docs: update CONTRIBUTING.md

Large documentation update to CONTRIBUTING.md which has absorbed and expanded upon the previous version to help improve the gliff.ai GitHub Community Health Files.

joshuajames-smith committed:
style: add punctuation

Add full stop to a line missing punctuation for consistency and reading purposes.

joshuajames-smith committed:
fix: navigation link

Linting, Formatting and Testing TOC fix so the navigation works.

joshuajames-smith committed:
fix: ROADMAP repository navigation link

Fix URL of ROADMAP repository navigation link so it now works.

joshuajames-smith committed:
style: bold text lines

Make emoji text lines in introduction bold to match styling of other community documentation.

joshuajames-smith committed:
docs: remove duplicated line

Duplicate style lines in commit message conventions list removed.

joshuajames-smith committed:
style: H2 for sub-headings

Document heading now H1 `#` and all other sub-headings that follow are now H2 `##`.

joshuajames-smith committed:
feat: add subject to mailto link

Add subject `[GitHub]` to email mailto link.

joshuajames-smith committed:
style: line break in contact

Add line break in `Contact` section for style consistency across documentation.

joshuajames-smith committed:
style: remove duplicated word

Removed the duplication of `repository repository` in `Labels` section.

joshuajames-smith committed:
feat: add links to other documentation

Added two links for CODE_OF_CONDUCT.md (anticipating location address, dependant on approval of issue #3) and SECURITY.md.

joshuajames-smith committed:
style: remove spacing

Remove double space styling for better whole text consistency.

joshuajames-smith committed:
feat: add links to gliff.ai website

Add links to any mention of `gliff.ai` in the introductions body text to offer link to website.

ChasNelson1990 commented on CONTRIBUTING.md:
```diff
@@ -1,52 +1,160 @@
+Each repository has one of two topic tags to help you and fellow contributors understand which repositories [gliff.ai](https://gliff.ai) is open to contributions outside of our team. They are:
+
+- **contributors-welcome** – We welcome contributions on this repository!
+- **no-contributors** – This repository does not currently accept contributions unfortunately!
```
But we still accept Issues if you see a problem?

ChasNelson1990 commented on CONTRIBUTING.md:
```diff
@@ -1,52 +1,160 @@
+Our team uses 4 templates for creating issues across gliff.ai repositories, and we would appreciate you also using these and filling them out with as much detail as you are able to provide. The templates include:
+
+- **Bug Report** – report any bugs whether technical or visual (the “bug” label will be automatically applied). 
+- **Feature/Enhancement Request** – request any new features or enhancements of current features (the “enhancement” label will be automatically applied).
```
Should we actually have a separate "New feature" vs "Enhancement" process?

joshuajames-smith replied:
I had this thought as it would be worth that extra organisation level with another label. I'll get on it separating this into two issue templates and auto-assigned labels!

ChasNelson1990 commented on CONTRIBUTING.md:
```diff
@@ -1,52 +1,160 @@
+- **Bug Report** – report any bugs whether technical or visual (the “bug” label will be automatically applied). 
+- **Feature/Enhancement Request** – request any new features or enhancements of current features (the “enhancement” label will be automatically applied).
+- **Test Request** – request a new test not currently available (the “test” label will be automatically applied).
+- **Security Vulnerability** – report a suspected vulnerability which risks the security of our code (the “vulnerability” label will be automatically applied, and two gliff.ai team members will be automatically assigned). Read the [gliff.ai Security Policy](https://github.com/gliff-ai/.github/blob/main/SECURITY.md) to learn more!
```
@cooper667 - perhaps we shouldn't have public vulnerability issues in repos? What do you think? We could just have the security policy ask them to email "security@gliff.ai" and have that auto message #techteam in Slack?

cooper667 replied:
Yea, pretty much. Don't even need the Slack really, just have it email you as it should be super rare except for spammers?

ChasNelson1990 replied:
I have had it just email you @cooper667  - because it'll get lost in my inbox!

ChasNelson1990 commented on CONTRIBUTING.md:
```diff
@@ -1,52 +1,160 @@
+
+[{{back to navigation}}](#table-of-contents)
+
+The **Pull Request** template should be filled out as much as possible. Your pull request should include a summary of the changes you have made and your motivation behind them. Please list any dependency changes, added tests, documentation changes, migrations and complete our checklist to ensure you have completed all the necessary steps.
```
"migrations" -> "database changes (if applicable)"?

ChasNelson1990 commented on CONTRIBUTING.md:
```diff
@@ -1,52 +1,160 @@
+
+The **Pull Request** template should be filled out as much as possible. Your pull request should include a summary of the changes you have made and your motivation behind them. Please list any dependency changes, added tests, documentation changes, migrations and complete our checklist to ensure you have completed all the necessary steps.
+
+Please remember to first determine which repository the pull request should be created in, then [search the gliff.ai repositories](https://github.com/search?q=user%3Agliff-ai+is%3Apr+is%3Aopen) to ensure the pull request you want to open is not already open! 
```
.... or, even better, closed!

ChasNelson1990 commented on CONTRIBUTING.md:
```diff
@@ -1,52 +1,160 @@
 
-We use Pre-commit for running lints, tests and other checks locally on commit and push.
-To use pre-commit on your own machine just run:
+We use pre-commit for running lints, tests and other checks locally on commit and push. To use pre-commit on your own machine, you should run the following in your terminal:
```
We haven't used pre-commit for about 7 months lol. It's all done in GitHub Actions now.

ChasNelson1990 commented on CONTRIBUTING.md:
```diff
@@ -1,52 +1,160 @@
 
-We are using pytest for our Python tests and jest with react-testing-library for our TypeScript testing.
-Configuration files can be found in our repos.
+Our teams uses flake8 and black for linting and auto-formatting our Python code alongside pytest for our Python tests, and then eslint and prettier.io for linting and auto-formatting our TypeScript code alongside jest with react-testing-library for our TypeScript testing. Further specifics and configuration files can be found in each of our repositories.
```
We should add something about BrowserStack too? Or perhaps that should be explained on a repo by repo basis?

joshuajames-smith replied:
I was about to say possibly a repo by repo basis? as these are more universal or widely used (but then its one or the other code languages) – so maybe it's worth mentioning here and repo by repo anyways?

ChasNelson1990 commented on CONTRIBUTING.md:
```diff
@@ -1,52 +1,160 @@
+
+Commit messages should indicate the type and include a short summary of the commit using the following conventions:
+
+- **feat**: The new feature you're adding to a particular application.
```
Hmmm... as a team we tend to use `chore` to include `style` and we also use `build`.

ChasNelson1990 commented on CONTRIBUTING.md:
```diff
@@ -1,52 +1,160 @@
-Thanks,
-The gliff.ai team
+Need some help? 🤔 Have a question? 🧠 \
+Reach out to the gliff.ai team at [contact@gliff.ai](mailto:contact@gliff.ai?subject=[GitHub]) or on GitHub.
```
Maybe this should be a #techteam shared "community@gliff.ai" and, again, be autoforwarded to Slack?

joshuajames-smith replied:
I think that's a better idea! I'll change emails across documentation now. Do we want the autoforwarded messages to be in a separated "github" channel for slack? As we did with other alerts etc?

ChasNelson1990 replied:
Yes, probably. Let's touch base on emails/slack integration stuff tomorrow.

joshuajames-smith committed:
fix: change contact email

Change contact email.

joshuajames-smith committed:
docs: change commit messaging

Remove `style` to integrate with `chore` to fit with the teams current practices. Added `build` as this is also used.

joshuajames-smith committed:
docs: update pr wording

Added in new link for user to check new pr is both not open or closed.

joshuajames-smith committed:
docs: change pr wording

Changed `migrations` to `database changes` in pr description.

joshuajames-smith committed:
docs: remove wording around pre-commit

In section `Linting, Formatting and Testing`, wording around `pre-commit` has been removed as this is outdated – now handled by GitHub Actions. Added links.

joshuajames-smith committed:
docs: added browerstack testing

Added in section `Linting, Formatting and Testing` about BrowerStack testing. Might not have wrote about it in the best way, open to suggestions/changes @ChasNelson1990!

joshuajames-smith committed:
docs: change contribution wording

Mentioned in #1 - added to `no-contributions` that issues are accepted if a user spots a problem and wants to raise attention to it. Thought about changing topic wording to `issue-only-contributions` but thought this might bring attention to repos that we are okay with no main contribution types (@ChasNelson1990 thoughts?).

joshuajames-smith committed:
docs: add new label

New `feature` label added as feature and enhancement templates are being separate rather than kept as one issue template.

joshuajames-smith committed:
docs: include new issue templates to list

Previous `feature/enhancement request` template is now two separate templates as `feature request` and `enhancement request`. Updated text in `issue` section to reflect this.

joshuajames-smith committed:
docs: update security report process

Remove security issue template in place of the new process of reporting via email with further details found in the `gliff. Security Policy`.

joshuajames-smith committed:
docs: update vulnerability label

Update `vulnerability` label description to better suit removal of security issue template as report process.

joshuajames-smith committed:
chore: add to `Table of Contents`

joshuajames-smith committed:
docs: remove inactive labels

Removed `ui/ux` and `documentation` labels as this is now handled by project board.

joshuajames-smith committed:
:wrench: chore: update label section

Removed `status` category as this is now replaced by new github (beta) project board metadata. Remove `javascript` and `vulnerability` labels. Adjust label category colours.
